src/DynamoCore/Extensions/ExtensionManager.cs

```diff
@@ -8,6 +8,40 @@
 
 namespace Dynamo.Extensions
 {
+
+    /// <summary>
+    /// An object which may request extensions to be loaded and added to the extensionsManager
+    /// </summary>
+    public interface IExtensionSource
+    {
+        /// <summary>
+        /// event that is raised when package manager requests an extension be loaded
+        /// which was found within a package it is currently loading.
+        /// </summary>
```

Comment:
This `event` is more general than the comment suggests. No?

Is it idiomatic to suffix an `event` with `Handler`? https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members#names-of-events